pkcs11/_pkcs11.pyx

```diff
@@ -253,6 +253,31 @@ class Slot(types.Slot):
 class Token(types.Token):
     """Extend Token with implementation."""
 
+    def init_token(self, token_label, so_pin):
+        cdef CK_SLOT_ID slot_id = self.slot.slot_id
+        cdef CK_UTF8CHAR *pin_data
+        cdef CK_ULONG pin_length
+        cdef CK_UTF8CHAR *label
+
+        if token_label is None or so_pin is None:
+            raise ArgumentsBad("Set both `token_label` and `so_pin`")
+
+        pin = so_pin.encode('utf-8')
+        tlabel = token_label.encode('utf-8')
+
+        if pin and tlabel:
+            pin_data = pin
+            pin_length = len(pin)
+            label = tlabel
+
+            with nogil:
+                assertRV(_funclist.C_InitToken(slot_id, pin_data, pin_length,
+                                               label))
+
+            return True
```

Comment:
Rather than returning True/False out of this function, it's probably better to return `None` or `raise ArgumentsBad`. That means you can try/except/finally your codepath.